plugins/sftpsend/main.py

```diff
@@ -0,0 +1,160 @@
+#!/usr/bin/env python3
+
+import paramiko
+import datetime
+import time
+from mfutil import get_unique_hexa_identifier
+from acquisition.step import AcquisitionStep
+
+
+class SftpSendStep(AcquisitionStep):
+
+    def init(self):
+        self._session = None
+        self._session_dt = None
+        self._logger = self._get_logger()
+        self.sftp_hostname = self.get_custom_config_value("sftp_hostname",
+                                                         default="FIXME")
+        self.sftp_username = self.get_custom_config_value("sftp_username",
+                                                         default="FIXME")
+        self.sftp_password = self.get_custom_config_value("sftp_password",
+                                                         default="FIXME")
+        if self.sftp_hostname == "FIXME" or self.sftp_hostname == "null" \
+                or self.sftp_hostname == "":
+            raise Exception("you have to set a valid sftp_hostname value")
+        if self.sftp_username == "FIXME" or self.sftp_username == "null" \
+                or self.sftp_username == "":
+            raise Exception("you have to set a valid sftp_username value")
+        if self.sftp_password == "FIXME" or self.sftp_password == "null" \
+                or self.sftp_password == "":
+            raise Exception("you have to set a valid sftp_password value")
```

Comment:
I don't know SFTP very much but if you use a key_file, you don't need password, do you?

Reply:
OK fixed